src/uwtools/config.py

```diff
@@ -0,0 +1,198 @@
+'''
```

Comment:
I can see that this is going to become unmanageable.  Even with the "dictable" types that have been added, it is already too big.

Reply:
I'm not sure it would necessarily become unmanageable if we limit it to only parse standard file types. All the ones that fall outside that will use templates as is the current way.

We already need a way to parse YAML and bash configure files at a basic workflow level. This really just adds f90nml to the mix beyond that, and that is something that I feel will allow the Apps to couple more directly with the model regression tested files. The goal moving forward would hopefully be to persuade the developers of components to adopt _standard_ config languages to configure the UFS components so that we don't have an untenable number of languages to support in the workflow.

The proliferation of config types does need to be managed, for sure. We have the power in the UFS community not to let it get any worse at a component level!